src/RequestParser.php

```diff
@@ -79,41 +94,48 @@ private function createModelBuilderStruct(Request $request): ModelBuilderStruct
 
     private function getBaseModelName(Request $request): string
     {
-        $modelCandidates = [];
-        $route = $request->route();
-        $controller = $this->getControllerFromRoute($route);
-        if ($controller) {
-            $stringToRemove = "controller";
-            $className = class_basename($controller);
-            $maybeBaseModel = substr_replace($className, '', strrpos(strtolower($className), $stringToRemove), strlen($stringToRemove));
-            $modelCandidates[] = $maybeBaseModel;
-
-            $modelName = $this->getModelFromNamespaces($maybeBaseModel, $this->modelNamespaces);
-            if ($modelName) {
-                return $modelName;
-            }
+        if ($this->modelName) {
+            return $this->modelName;
         }
 
-        $exploded = explode("/", $request->getRequesturi());
-        $lastURISegment = strtolower(end($exploded));
-        $lastURINoQuery = current(explode("?", $lastURISegment, 2));
-        $camelizeURI = str_replace('_', '', ucwords($lastURINoQuery, '_'));
-        $modelCandidates[] = $camelizeURI;
+        $errorMessage = 'Model not found, please use "setModel()" method.';
+        if ($this->guessModelName) {
+            $modelCandidates = [];
```

Comment:
This code is so bloated on this method.
Please move it into another, method.
`getGuessModelName` perhaps ?

Reply:
> This code is so bloated on this method.

How if I help the refactor portion in other PR @LIQRGV? Because I believe this PR is just for our new model setter method. :grin: